src/main/java/com/google/cloud/datafusion/plugin/successfactors/source/metadata/SuccessFactorsSchemaGenerator.java

```diff
@@ -0,0 +1,770 @@
+   * @throws EdmException any apache olingo processing exception.
+   */
+  @Nullable
+  private SuccessFactorsColumnMetadata buildNavigationColumns(String entityName, String navPath) throws EdmException {
+    EdmNavigationPropertyImplProv association = successFactorsServiceHelper.getNavigationProperty(entityName, navPath);
+    if (association == null) {
+      LOGGER.debug(ResourceConstants.DEBUG_NAVIGATION_NOT_FOUND.getMsgForKey(navPath, entityName));
+      return null;
+    }
+
+    EdmEntityType entitySet = successFactorsServiceHelper.extractEntitySetFromNavigationProperty(association);
+    if (entitySet == null) {
+      LOGGER.debug(ResourceConstants.DEBUG_NAV_PROP_NOT_FOUND.getMsgForKey(association.getName(), navPath));
+      return null;
+    }
+
+    List<String> propList = entitySet.getPropertyNames();
+    List<SuccessFactorsColumnMetadata> columns = buildSuccessFactorsColumns(entitySet, propList);
+
+    return SuccessFactorsColumnMetadata.builder()
+      .name(association.getName())
+      .type(entitySet.getKind().name())
+      .multiplicityOrdinal(association.getMultiplicity().ordinal())
+      .childList(columns)
+      .build();
+  }
+
+
+  /**
+   * Build schema for all the given selective property under the 'selectOption'.
+   *
+   * @param entityName   service entity name
+   * @param selectOption all the selective property names
+   * @return {@code Schema}
+   * @throws SuccessFactorsServiceException throws in following two cases
+   *        1. if no selective property were found in the given entity name,
+   *        2. if fails at apache olingo processing.
+   */
+  public Schema buildSelectOutputSchema(String entityName, String selectOption) throws SuccessFactorsServiceException {
+    try {
+      List<SuccessFactorsColumnMetadata> columnDetailList = buildSelectedColumns(entityName, selectOption);
+      if (columnDetailList.isEmpty()) {
+        throw new SuccessFactorsServiceException(ResourceConstants.ERR_NO_COLUMN_FOUND.getMsgForKey(selectOption,
+                                                                                                    entityName));
+      }
+
+      freezeSuccessFactorsColumnMetadata(columnDetailList);
+
+      return buildSchema(columnDetailList);
+    } catch (EdmException ee) {
+      throw new SuccessFactorsServiceException(
+        ResourceConstants.ERR_BUILDING_COLUMNS.getMsgForKey(selectOption, entityName), ee);
+    }
+  }
+
+  /**
+   * Finds and builds all the selective (navigation & non-navigation) property provided under 'selectEntity'
+   *
+   * @param entityName  service entity name
+   * @param selectEntry selective property name
+   * @return list of {@code SuccessFactorsColumnMetadata} or empty list in case of invalid selective property name.
+   * @throws EdmException any apache olingo processing exception.
+   */
+  private List<SuccessFactorsColumnMetadata> buildSelectedColumns(String entityName, String selectEntry)
```

Comment:
By columnMetadata do we mean the contents of the column or does it refer to other metadata associated with the column?

Reply:
The metadata associated with that column like type, length, etc.